hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -637,7 +632,12 @@ class Resolver(tl: TraceLogger)
             case t: Term.Ref => sym.map(sym => t.resSym = S(sym))
           log(s"Resolved symbol for ${t}: ${lhsDefn.sym}")
         case _ =>
-    case _ => 
+    case _ =>
+    
+    t match
+    case t: Term.Ref if t.resSym.isEmpty =>
```

Comment:
How about you document what's going on here?

From what I can surmise, it's something like
```suggestion
    // If a reference was not resolved to take implicit arguments or return some module type,
    // then its result symbol is the same as the symbol it refers to.
    case t: Term.Ref if t.resSym.isEmpty =>
```